test/compile_and_run_test.sh

```diff
@@ -59,7 +59,9 @@ compileAndRunVersion()
 
 testReleased6Versions()
 {
+  echo "deb [trusted=yes] https://apt.datadoghq.com/ stable 6" > "${BUILDPACK_HOME}/etc/datadog.list"
```

Comment:
Why adding a line with `trusted=yes` then replacing it with `signed-by`?

Reply:
The test first gets available versions, before running the `bin/compile` script. This means that at this point the GPG keyring is not created yet and so we have to skip repodata verification at this point. After we get the list of versions, we revert the source list file to the previous state and then run `bin/compile` (by running `compileAndRunVersion`), which does its own thing, now while actually using `signed-by`.